server/index.js

```diff
@@ -40,7 +40,7 @@ app.get('/api/products/:productId', (req, res, next) => {
   db.query(sql, params)
     .then(result => {
       const product = result.rows[0];
-      if (!product) {
+      if (!Number.isInteger(product)) {
```

Comment:
The code you had here was fine. You should do this check before querying the database since you can. It's a bad request (400) if the client sent a bad value.